apps/heureka/src/App.tsx

```diff
@@ -22,22 +21,48 @@ export type AppProps = {
   apiEndpoint?: string
   embedded?: boolean
   initialFilters?: InitialFilters
+  basePath?: string
+  enableHashedRouting?: boolean
 }
 
-const App = (props: AppProps) => (
-  <ApolloProvider client={getClient({ uri: props.apiEndpoint })}>
-    <AppShellProvider theme={`${props.theme ? props.theme : "theme-dark"}`}>
-      {/* load styles inside the shadow dom */}
-      <style>{styles.toString()}</style>
-      <ErrorBoundary>
-        <StoreProvider
-          initialState={{ selectedView: { viewId: UserView.Services }, initialFilters: props?.initialFilters }}
-        >
-          <Shell {...props} />
-        </StoreProvider>
-      </ErrorBoundary>
-    </AppShellProvider>
-  </ApolloProvider>
-)
+// Static router just for type registration
+const staticRouter = createRouter({
+  routeTree,
+  context: {
+    appProps: undefined!,
+  },
+})
+
+declare module "@tanstack/react-router" {
+  interface Register {
+    router: typeof staticRouter
+  }
+}
+
+const App = (props: AppProps) => {
+  const router = React.useMemo(
```

Comment:
why do you need to useMemo here?

Reply:
So we don't recreated router whenever App re-renders but that might not be needed because the `appProps` do not change dynamically although they could looking from the Greenhouse/shell app perspective.